test/e2e/pages/new-lex-project.page.ts

```diff
@@ -0,0 +1,147 @@
+import { expect, Locator, Page } from '@playwright/test';
+
+type ChooserPage = {
+  sendReceiveButton: Locator;
+  createButton: Locator;
+};
+
+type NamePage = {
+  projectNameInput: Locator,
+  projectCodeInput: Locator,
+  projectCodeUneditableInput: Locator,
+  projectCodeLoading: Locator,
+  projectCodeExists: Locator,
+  projectCodeAlphanumeric: Locator,
+  projectCodeOk: Locator,
+  editProjectCodeCheckbox: Locator
+};
+
+type SrCredentialsPage = {
+  loginOk: Locator,
+  loginInput: Locator,
+  passwordInput: Locator,
+  credentialsInvalid: Locator,
+  passwordOk: Locator,
+  projectNoAccess: Locator,
+  projectOk: Locator
+  projectSelect: Locator
+};
+
+type VerifyDataPage = {
+  title: Locator,
+  nonCriticalErrorsButton: Locator,
+  entriesImported: Locator,
+  importErrors: Locator,
+};
+
+type SelectLanguageModal = {
+  searchLanguageInput: Locator,
+  languageRows: Locator,
+  addButton: Locator,
+};
+
+export class NewLexProjectPage {
+  readonly page: Page;
+
+  readonly newLexProjectForm: Locator;
+  // form controls
+  readonly backButton: Locator;
+  readonly nextButton: Locator;
+  readonly formStatus: Locator;
+  readonly progressIndicatorStep3Label: Locator;
+
+  // step 0: chooser
+  readonly chooserPage: ChooserPage;
+
+  // step 1: project name
+  readonly namePage: NamePage;
+
+  // step 1: send receive credentials
+  readonly srCredentialsPage: SrCredentialsPage;
+
+  // step 2: initial data
+  readonly initialDataPageBrowseButton: Locator;
+
+  // step 3: verify data
+  readonly verifyDataPage: VerifyDataPage;
+
+  // step 3 alternate: primary language
+  readonly primaryLanguagePageSelectButton: Locator;
+  // select language modal
+  readonly selectLanguage: SelectLanguageModal;
+
+  static readonly url: string = '/app/lexicon/new-project';
+
+  constructor(page: Page) {
```

Comment:
I think the page objects would be simpler to read if we declared and assigned at the same time. Also, I'd then see the value of the locator when I navigate to the field, rather than having to find the assignment down in the constructor.

E.g.
```
readonly newLexProjectForm = this.page.locator('#new-lex-project-form');

constructor(readonly page: Page) {
}
```